### PR TITLE
remove mocking framework 'mocha'

### PR DESCRIPTION
**Pull request by linki**
This PR removes `mocha` as a dependency from OpenProject (Core).

Numerous times I got confused when using `stub`, `stubs`, `returns`, `and_return`, `expects`, `should_receive` etc. due to rspec and mocha providing mocking facilities. Today, I ran into some trouble where a mocha mock wasn't removed correctly. Since nowadays `rspec-mocks` provides almost the same mocking capabilities as mocha does, I decided to remove the last remaining bits of it and exclusively use rspec-mocks in OpenProject's core.

note to reviewer: second commit does indentation only.

### CODE REVIEW

File: test/unit/comment_test.rb [hunk 1]
```diff
@@ -81,13 +81,13 @@ def test_create_should_send_notification_with_settings
     news = FactoryGirl.create(:news, :project => project, :author => user)
 
     # with notifications for that event turned on
-    Notifier.stubs(:notify?).with(:news_comment_added).returns(true)
+    Notifier.stub!(:notify?).with(:news_comment_added).and_return(true)
```
Comment (ulferts) [inline test/unit/comment_test.rb]: using stub! is deprecated in favor of stub.

**Comment by ulferts** [comment]
I like it. :+1: 

Let's see if the plugins do also. They might rely on mocha.

**Comment by GarstgerUnhold** [comment]
@ulferts Some plugins do rely on mocha. At the very least mobile_otp and video_upload. Check out the PRs in those plugins as well.

**Comment by ulferts** [comment]
@GarstgerUnhold thx for pointing me to it. I still have to get used to the new github design.

**Pull request merged**